addons/misc/functions/fnc_suitcaseNukeZeus.sqf

```diff
@@ -0,0 +1,211 @@
+#include "script_component.hpp"
+/*/////////////////////////////////////////////////
+Author: Landric
+               
+File: fnc_suitcaseNukeZeus.sqf
+Parameters: pos
+Return: none
+
+*///////////////////////////////////////////////
+params [["_pos",[0,0,0],[[]],3]];
+
+private _onConfirm =
+{
+    params ["_dialogResult","_in"];
+    _in params [["_pos",[0,0,0],[[]],3]];
+    _dialogResult params [
+        "_countdownLength",
+        "_effect",
+        "_defuser",
+        "_defuseTime",
+        ["_code", "", [""]],
+        ["_target", 0, [0]]
+    ];
+
+    private _device = "Land_MultiScreenComputer_01_black_F" createVehicle _pos;
+    ["zen_common_updateEditableObjects", [[_device]]] call CBA_fnc_serverEvent;
+    _device setVariable [QGVAR(suitcaseNuke_isArmed), true, true];
+
+
+    // Timer code
+    // TODO: move to server to avoid problems if zeus crashes?
```

Comment:
Sadly can't assume server would have this loaded as clientside only. So no real way to offload custom functions easily. Only option would remoteExec a spawn, but unsure how it transfers the code in the spawn.

Reply:
Yes, remoteExec was the approach I was considering, but that brings its own set of issues. Probably best to leave it on the zeus's machine; but worth testing exactly what happens if the zeus leaves/disconnects (presumably the code just stops running and the device becomes inert - but it might be nice if there was a way of maintaining it or cleaning it up!)

Reply:
What about an EH for if/when the object changes locality?

Reply:
> What about an EH for if/when the object changes locality?

on which client would you register said EH? It won't fire on the Zeus' machine if already left ;-)

Hmm, does ZEN or CBA not have a event you can call that spawns or executes code? Might be able to piggy-back on that to send code to run on server. Like the execute_code module and sending the code as string? Not pretty, but this mod require CBA and ZEN loaded on server, so it should be available?

Reply:
> on which client would you register said EH? It won't fire on the Zeus' machine if already left ;-)
If registered on all machines, it'll only fire on the one it was transferred from (except they've left), and whichever one it moves to (which could in theory take over the timer/activation)

I'm not sure its a *good* idea...

> Hmm, does ZEN or CBA not have a event you can call that spawns or executes code?
Maybe; the reason not to do a straight-up `remoteExec` is because some servers/clients will block use of `call` with arbitrary code, and I'd be surprised if CBA let you get around that

Reply:
> Maybe; the reason not to do a straight-up `remoteExec` is because some servers/clients will block use of `call` with arbitrary code, and I'd be surprised if CBA let you get around that

Pretty sure CBA gets around that, as you would send a CBA_EVENT which is received by a CBA_LISTENER on the other side. So its not a remoteExec. In general I believe I have seen places with disallowed RemoteExec work with CBA events instead. 

Although not completely sure about the underlying tech used for the actual communication.